src/foam/core/console/FileUpload.js

```diff
@@ -0,0 +1,1244 @@
+          start('h4').add('Or Manual Content Input').end().
+          start('p').addClass(this.myClass('text-muted')).
+            add('Paste your file content directly if not using file upload.').
+          end().
+          start('div').addClass('mt-lg').
+            start('label').add('Format: ').end().
+            add(this.FORMAT).
+            show(this.format$.map(f => f === 'CSV')).
+            start('label').addClass('ml-lg').add('Delimiter: ').end().
+            add(this.DELIMITER).
+          end().
+          start('div').addClass('mt-lg').
+            add(this.INPUT).
+          end().
+        end().
+      end().
+      
+      // Structure Analysis & DAO Selection
+      start('div').addClass(this.myClass('step')).
+        start('h3').add('Step 2: Analyze Structure & Select DAO').end().
+        start('div').addClass('mb-lg').
+          // Structure Analysis button
+          start('div').addClass('mt-lg').
+            start(this.ANALYZE_STRUCTURE).end().
+          end().
+          
+          // Structure Status and Results
+          start('div').addClass('mt-lg').
+            // Structure Analysis Results
+            show(this.filesVerified$).
+            start('div').addClass('p-sm').addClass(this.myClass('bg-light')).addClass('mb-lg').
+              start('h4').addClass('mb-sm').add('Structure Analysis Results').end().
+              start('div').addClass('mb-sm').addClass(this.myClass('text-success')).
+                add(this.uploadedFiles$.map(files => 
+                  `✅ Structure verified! All ${files ? files.length : 0} files have matching headers:`
+                )).
+              end().
+              start('div').addClass(this.myClass('text-mono')).addClass(this.myClass('bg-white')).addClass('p-sm').
+                add(this.detectedHeaders$.map(headers => headers.join(', '))).
+              end().
+            end().
+
+            // Target DAO Selection (only show after analysis)
+            show(this.filesVerified$).
+            start('div').addClass('mt-lg').
+              start('h4').add('Target DAO').end().
+              start('p').addClass(this.myClass('text-muted')).
+                add('Select the target DAO where the data will be uploaded.').
+              end().
+              start('div').addClass('mt-sm').
+                add(this.SELECTED_DAO).
+              end().
+            end().
+          end().
+        end().
+      end().
+      
+      // Mapping Review
+      start('div').addClass(this.myClass('step')).
+        start('h3').add('Step 3: Review Field Mappings').end().
+        start('div').
+          show(this.mappings$.map(mappings => mappings.length > 0)).
+          add(this.MAPPINGS).
+        end().
+        start('div').
+          show(this.mappings$.map(mappings => mappings.length === 0)).
+          start('p').add('No mappings generated yet. Please analyze the structure first.').end().
+        end().
+      end().
```

Comment:
Same here, can be handled only with property configuration and
```suggestion
          tag(this.MAPPINGS.__).
        
```